std/file.d

```diff
@@ -2287,21 +2287,21 @@ if (isConvertibleToString!R)
 // Same as mkdir but ignores "already exists" errors.
 // Returns: "true" if the directory was created,
 //   "false" if it already existed.
-private bool ensureDirExists(in char[] pathname)
+private bool ensureDirExists(in char[] pathname) @safe
 {
     import std.exception : enforce;
 
     version(Windows)
     {
-        if (CreateDirectoryW(pathname.tempCStringW(), null))
+        if (() @trusted { return CreateDirectoryW(pathname.tempCStringW(), null); }())
```

Comment:
I'm a bit uneasy on this, as you are trusting the pathname's range operations when you do `tempCStringW` (as well as assuming the range doesn't have such a member name and you aren't calling the wrong thing). Is it possible to do this outside the trusted lambda?